proposals/002-text-utf-default.md

```diff
@@ -0,0 +1,156 @@
+
+-   Ben Gamari: integration with GHC
+
+-   The Cabal maintainers (fgaz, emilypi, mikolaj): integration with Cabal
+
+Progress will be reported on a weekly basis to the HF Technical Agenda
+Track, with Emily as support for Andrew.
+
+# Timeline
+
+We expect that this project will take roughly 6 months to fully
+complete: 3-4 months to complete the code implementation, performance
+testing, and unit testing, another 1-2 months to integrate with
+stakeholders and diagnose any potential issues with the migration.
+
+**Preparation:**
+
+Using the HVR's existing [`text-utf8`](https://github.com/text-utf8) as
+a starting point, the following must be done before an implementation is
+started:
+
+-   Modernize the codebase and clear out the bitrot
+
+-   Establish a baseline for performance and any related issues.
+
+-   Update testing and performance benchmarks to make use of `inspection-testing` to ensure fusion is not broken in
+    subsequent UTF-8 related changes.
+
+An MVP should completely preserve standard user-facing API, and not
+break fusion. Performance should not significantly diverge from the
+existing UTF-16 text package. There will be an expected change to the
+exposed Text internals, in which case, breakage should be assessed by
+circulating a git commit reference to a release candidate as soon as
+possible. This candidate should be sourced publicly and loudly.
+
+**Implementation:**
+
+-   TBD: There is a straightforward implementation, but this one is left up to Andrew for comment.
+
+**Stakeholders:**
+
+-   Library authors will need to be made aware of changes and adjust accordingly. HF will provide a git reference to a complete MVP as
+    soon as possible, and produce a migration guide.
+
+-   In the case of GHC, Cabal, and other core infrastructure, we will work closely with these packages to help migrate and assess+fix
+    breakages.
+
+While we do not expect many authors to experience significant changes,
+there will be some help that needs to be given in terms of bumping
+Hackage bounds since this migration will be a major version bump. HF
+will need to coordinate with the Hackage Trustees to help move along
+packages that go out of date.
+
+# Deliverables
+
+-   text-2.0.0.0, which will provide a UTF-8 encoding for Text as a default for all versions going forward.
+
+-   A `text-utf16` package, which is a preservation of the current UTF-16 encoded text, for backwards compatibility.
+
+-   Updates to the Text Haddocks that reflect the UTF-8 changes
+
+-   Announcements and updates across all Haskell channels covering the following:
+    -   Significant dates and milestones
+
+    -   Expected code impact
+
+    -   Release candidates
+
+    -   Delivery
+
+-   Migration instructions and design documentation
+
+# Outcomes
+
+-   Addresses a recognized need and want from both Industry and the Haskell Community
+
+-   Better UX for Haskell's text story
```

Comment:
How so? This bullet implies a change to the API of `text`, but other commentary here says that the interface will not change (except for exported internals).

Reply:
I might be totally confused and wrong about this, but I was under the impression the changes outlined in this proposal (amongst other results) open the door / hallway towards improving the situation with `String`/`Text`/`ByteString` and the prolific use of `String` in our "core libraries" (despite the prolific recommendation to use `Text`/`ByteString` instead of `String`.

Reply:
As stated in "Motivation" section, at the moment people often use `ByteString` instead of `Text` when looking for ultimate performance. It's easy to convince yourself that you application will never encounter non-ASCII data, if it promises a performance boost. By backing `Text` with UTF-8, we improve performance of ASCII strings and thus assist convergence to using `Text` for anything which is text semantically (and keeping `ByteString` for binary data).